src/python/pants/backend/scala/target_types_test.py

```diff
@@ -0,0 +1,151 @@
+from pants.engine.internals.parametrize import Parametrize
+from pants.engine.rules import QueryRule
+from pants.engine.target import Target
+from pants.jvm import jvm_common
+from pants.jvm.target_types import (
+    JvmArtifactArtifactField,
+    JvmArtifactExclusionRule,
+    JvmArtifactGroupField,
+    JvmArtifactResolveField,
+    JvmArtifactTarget,
+    JvmArtifactVersionField,
+)
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    rule_runner = RuleRunner(
+        target_types=[ScalaArtifactTarget, JvmArtifactTarget],
+        rules=[
+            *target_types_rules(),
+            *jvm_common.rules(),
+            QueryRule(_TargetParametrizations, [_TargetParametrizationsRequest]),
+        ],
+        objects={
+            "parametrize": Parametrize,
+            JvmArtifactExclusionRule.alias: JvmArtifactExclusionRule,
+            ScalaArtifactExclusionRule.alias: ScalaArtifactExclusionRule,
+        },
+    )
+    return rule_runner
+
+
+def assert_generated(
+    rule_runner: RuleRunner,
+    address: Address,
+    *,
+    build_content: str,
+    scala_versions_per_resolve: dict[str, str],
+    expected_targets: set[Target],
+) -> None:
+    rule_runner.write_files({"BUILD": build_content})
+    rule_runner.set_options([f"--scala-version-for-resolve={repr(scala_versions_per_resolve)}"])
+
+    parametrizations = rule_runner.request(
+        _TargetParametrizations,
+        [
+            _TargetParametrizationsRequest(
+                address,
+                description_of_origin="tests",
+            ),
+        ],
+    )
+    assert expected_targets == {
+        t for parametrization in parametrizations for t in parametrization.parametrization.values()
+    }
+
+
+def test_generate_jvm_artifact_based_on_resolve(rule_runner: RuleRunner) -> None:
+    assert_generated(
+        rule_runner,
+        Address("", target_name="test"),
+        build_content=dedent(
+            """\
+            scala_artifact(
+                name="test",
+                group="com.example",
+                artifact="example-gen",
+                version="3.4.0",
+                resolve="current",
+            )
+            """
+        ),
+        scala_versions_per_resolve={"current": "2.13.10"},
+        expected_targets={
+            JvmArtifactTarget(
+                {
+                    JvmArtifactGroupField.alias: "com.example",
+                    JvmArtifactArtifactField.alias: "example-gen_2.13",
+                    JvmArtifactVersionField.alias: "3.4.0",
+                    JvmArtifactResolveField.alias: "current",
+                },
+                Address(
+                    "",
+                    target_name="test",
+                    generated_name="example-gen_2.13",
+                ),
+            ),
+        },
+    )
+
+
+def test_generate_jvm_artifacts_for_parametrized_resolve(rule_runner: RuleRunner) -> None:
```

Comment:
This test keeps failing because, even though it generates the number of expected targets (based on the fact that one field is parametrized), it always _sees_ the same value for the `resolve` field, resulting in the wrong suffix added to the `artifact` field.

I may be missing something in the implementation of the rule.